File: boj/b_13913.py
import sys
sys.stdin = open("input.txt", "r") # 제거
input = sys.stdin.readline

n, k = map(int, (input().split()))
from collections import deque
q = deque()
q.append((n, 0, [n]))
arr = [0] * 200001
while q:
    now, cnt, passed = q.popleft()
    if n > k:
        print(n - k)
        for i in range(n, k - 1, -1):
            print(i, end=" ")
        break
    if now == k:
        print(cnt)
        print(*passed)
        break
    for i in [now - 1, now + 1, now * 2]:
        if i < 0 or i > 100000:
            continue
        if not arr[i]:
            arr[i] = 1
            q.append((i, cnt + 1, passed + [i]))

File: boj/b_13913.js
function solution(n, k) {
  let ptr = -1;
  const q = [[n, 0, [n]]];
  const arr = Array(200001).fill(0);
  let resultCnt = 0;
  let resultPassed = [];
  while (true) {
    const [now, cnt, passed] = q[++ptr];
    if (n > k) {
      resultCnt = n - k;
      for (let i = n; i >= k; i -= 1) {
        resultPassed.push(i);
      }
      break;
    }
    if (now === k) {
      resultCnt = cnt;
      resultPassed = passed;
      break;
    }
    for (const i of [now - 1, now + 1, now * 2]) {
      if (i < 0 || i > 100000) continue;
      if (!arr[i]) {
        arr[i] = 1;
        q.push([i, cnt + 1, passed.concat(i)]);
      }
    }
  }
  return [resultCnt, resultPassed];
}
console.log(solution(5, 17));
